src/relations/peers.py

```diff
@@ -222,7 +221,8 @@ def _on_changed(self, event: RelationChangedEvent):
             try:
                 # raises an error if this is fired before on_pebble_ready.
                 self.charm.reload_pgbouncer()
-            except ConnectionError:
+                self.charm.toggle_monitoring_layer(self.charm.backend.ready)
+            except (ConnectionError, ChangeError):
```

Comment:
If the `listen_port` is changed, pebble cannot know and the monitoring service may fail before a PGB service is available on the new port.